### ISSUE TITLE
Capturing Catch-all Segments in Deep Linking

### ISSUE THREAD

**Issue by Crisfole** [react-navigation/react-navigation#9049]
**Current Behavior**

There currently appears to be no way to capture catchall segments as named props when [configuring links](https://reactnavigation.org/docs/configuring-links).

**Expected Behavior**

Like most other routing tools it'd be nice to be able to capture the rest of a route in a named prop. For example:

```js
{
  Settings: "settings",
  Home: "*rest",
}
```

In the Home component I'd expect to be able to do the following:

```js
export const Home = ({route}) => (
  <Text>The rest of the route is: {route.params.rest}</Text>
);
```

**How to reproduce**

N/A

**Comment by SleeplessByte** [comment]
FYI, if you need this now, you can do this by manually parsing via [`getStateByPath`](https://reactnavigation.org/docs/configuring-links/#advanced-cases)

**Comment by Crisfole** [comment]
@SleeplessByte yes! That's what we do at the moment, this was a ergonomic request.

**Comment by samuelpetroline** [comment]
@Crisfole can you explain how you achieved this? I have the same situation but I have no idea what to do haha

**Comment by SleeplessByte** [comment]
My comment links to the documentation @samuelpetroline. Perhaps that helps?

**Comment by samuelpetroline** [comment]
@SleeplessByte it does but that's not the problem. My config has deeply nested routes and I can't directly access the state `params` because of this. 

![image](https://user-images.githubusercontent.com/22039767/110548558-832a1d80-810f-11eb-8ccf-fdf3568e5a33.png)

I was just wondering if there was any way to change the `params` only. I guess the best option is to recursively iterate until `params` is found and then change

**Comment by SleeplessByte** [comment]
That's the way to do it.

**Comment by samuelpetroline** [comment]
@SleeplessByte thank you, gonna try this way!
